python/ray/data/tests/mock_http_server.py

```diff
@@ -0,0 +1,80 @@
+# extracted from aioboto3
```

Comment:
I will rename the [existing s3 mock server](https://github.com/ray-project/ray/blob/master/python/ray/data/tests/mock_server.py) from mock_server.py to mock_s3_server.py later in another PR. Do not want to pack the renaming change here to make review easier.